alfasim_sdk/data_types.py

```diff
@@ -0,0 +1,94 @@
+import numbers
+from typing import List
+
+import attr
+from attr import attrib
+from attr.validators import instance_of, optional
+
+
+@attr.s
+class BaseField:
+    """
+    Base class for all widget fields available at alfasim.
+    """
+    caption: str = attrib(validator=instance_of(str))
+
+
+@attr.s
+class String(BaseField):
+    """
+    The String represents an input that the user can provide a string text for the application.
+
+    GUI Representation:
+    On the GUI interface, the String represent an text input with a one-line text editor.
+
+    Properties:
+    caption - property used as a label for the text input.
+    value   - property that holds the value informed from the user, or the default value that
+              should be displayed for the user.
+
+
+    """
+    value: str = attrib(validator=instance_of(str))
+
+
+@attr.s
+class Enum(BaseField):
+    value: List[str] = attrib()
+    initial: str = attrib(validator=optional(instance_of(str)), default=None)
+
+    @value.validator
+    def check(self, attr, values):
+        if not isinstance(values, (list, set)):
```

Comment:
Not sure we should allow `set`s, as they don't keep order.

Reply:
✔️ `set` removed